platform/server/elasticsearch/Cluster.ts

```diff
@@ -31,42 +30,8 @@ export class Cluster {
 
     this.log.info('cluster client stopped');
   }
-
-  // Attempt at typing the current `callWithRequest`, which is really difficult
-  // to type properly the way it is right now.
-  // TODO This should _return_ a client preset with the headers, but this
-  // requires changing Elasticsearch.js
-  async withRequest<T>(
-    req: KibanaRequest,
-    cb: (client: Client, headers: { [key: string]: any }) => T,
-    options: {
-      wrap401Errors: boolean;
-    } = {
-      wrap401Errors: false
-    }
-  ) {
-    const headers = req.getFilteredHeaders(this.config.requestHeadersWhitelist);
-
-    try {
-      return await cb(this.client, headers);
-    } catch (err) {
-      // instanceof err check?
-
-      if (options.wrap401Errors && err.statusCode === 401) {
-        console.log('401 + wrap', err);
-        throw err;
-
-        // From current Kibana:
-
-        // const boomError = Boom.wrap(err, err.statusCode);
-        // const wwwAuthHeader = get(err, 'body.error.header[WWW-Authenticate]');
-        // boomError.output.headers['WWW-Authenticate'] = wwwAuthHeader || 'Basic realm="Authorization Required"';
-
-        // throw boomError;
-      }
-
-      console.log('request failed yo', err);
-      throw err;
-    }
-  }
+  //TODO: removed withRequest, need to implement
+  //callWithRequest or something
+  // Should `callWithRequest`/`getCluster` even live here,
+  // or in ElasticsearchService.ts?
```

Comment:
I think `callWithRequest` should live here on the `Cluster`. So you can `getCluster` (or something like that) from the ElasticsearchService, which returns a cluster of the correct type (e.g. `data`, `admin`) that you can `callWithRequest` on. Then we can start adding more methods on this `Cluster` class later, like `search` etc.